frame/composable-support/src/types/mod.rs

```diff
@@ -4,6 +4,20 @@ use codec::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
 use sp_runtime::RuntimeDebug;
 
+/// Cosmos AccAddress type
+///
+/// Cosmos supports both secp256k1 & secp256r1 for transaction authentication.
+/// Public Keys for both are in the ECDSA 33-byte compressed format.
+#[derive(
+	Clone, Copy, Decode, Encode, Eq, Hash, MaxEncodedLen, PartialEq, RuntimeDebug, TypeInfo,
+)]
+pub enum CosmosAddress {
+	/// Address length will be 20 bytes long
+	Secp256k1([u8; 33]),
```

Comment:
Shouldn't this be `[u8; 20]` then?

Reply:
This name is misleading. The 33 bytes are the compressed public key, which a Cosmos AccAddress can be derived from.
Public key lengths are the same for both k1 and r1 signatures.

Reply:
dfdb3556c3de2b297282db6c1e29e81745bd1a53